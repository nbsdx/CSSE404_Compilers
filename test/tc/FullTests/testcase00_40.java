//Invalid code

class myMainClass{

	public static void main(String[] args){
		a = 5;	//undefined variable
		Getter get = new Getter();
		int c = get.set(a);
		int c = get.get();
	}
}

class Getter{

	int field;

	public int set(int x){
		int x = 4;
		field = x + 1;
		return x;
	}

	public int get(){
		return field;
	}
}