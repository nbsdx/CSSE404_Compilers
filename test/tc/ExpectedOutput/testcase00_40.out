No variable named a exists in the current scope.
Variable a is not declared.
Argument type null is incompatible with formal parameter type int
The variable c is already declared in the current scope.
The variable x is already declared in the current scope.
